packages/frontend-2/components/settings/workspaces/automation/Functions/Functions.vue

```diff
@@ -0,0 +1,117 @@
+          label: 'Open docs'
+        }
+      ]"
+    >
+      <p class="text-body-xs text-foreground-2 mt-2">
+        View and manage functions accessible only to projects in your workspace
+      </p>
+    </SettingsSectionHeader>
+    <LayoutTable
+      :columns="[
+        { id: 'name', header: 'Name', classes: 'col-span-3' },
+        { id: 'owner', header: 'Owner', classes: 'col-span-3' },
+        { id: 'id', header: 'ID', classes: 'col-span-5' },
+        { id: 'actions', header: '', classes: 'col-span-1 flex justify-end' }
+      ]"
+      :items="workspaceFunctions"
+    >
+      <template #name="{ item }">
+        <div class="flex items-center gap-2">
+          <AutomateFunctionLogo size="md" :logo="item.logo" />
+          <NuxtLink
+            class="text-foreground-3 hover:text-foreground-2 underline"
+            :to="automateFunctionRoute(item.id)"
+          >
+            {{ item.name }}
+          </NuxtLink>
+        </div>
+      </template>
+      <template #owner="{ item }">
+        <div class="flex items-center gap-2">
+          <UserAvatar
+            hide-tooltip
+            :user="item.creator"
+            light-style
+            class="bg-foundation"
+            no-bg
+          />
+          <span class="truncate text-body-xs text-foreground">
+            {{ item.creator?.name }}
+          </span>
+        </div>
+      </template>
+      <template #id="{ item }">
+        <div class="flex items-center text-foreground-2">
+          {{ item.id }}
+          <ClipboardIcon
+            class="w-4 h-4 ml-2 cursor-pointer hover:text-foreground transition"
+            @click="() => handleCopyText(item.id)"
+          />
+        </div>
+      </template>
+      <template #actions="{ item }">
+        <SettingsWorkspacesAutomationFunctionsTableRowActions
+          :workspace-function="item"
+          :permissions="item.permissions"
+        />
+      </template>
+    </LayoutTable>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { graphql } from '~/lib/common/generated/gql'
+import type { SettingsWorkspacesAutomationFunctions_WorkspaceFragment } from '~/lib/common/generated/gql/graphql'
+import { automateFunctionRoute } from '~/lib/common/helpers/route'
+import { ClipboardIcon } from '@heroicons/vue/24/outline'
+
+graphql(`
+  fragment SettingsWorkspacesAutomationFunctions_Workspace on Workspace {
+    id
+    automateFunctions(limit: 50, filter: { includeFeatured: false }) {
+      items {
+        id
+        name
+        logo
+        creator {
+          id
+          name
+          avatar
+        }
+        permissions {
+          canRegenerateToken {
+            authorized
+            message
+            code
+            payload
+          }
+        }
+        ...SettingsWorkspacesAutomationRegenerateTokenDialog_AutomateFunction
+      }
+    }
+  }
+`)
+
+defineProps<{
+  workspaceFunctions: SettingsWorkspacesAutomationFunctions_WorkspaceFragment['automateFunctions']['items']
```

Comment:
```
fragment SettingsWorkspacesAutomationFunctions_AutomateFunctionCollection on AutomateFunctionCollection {
  items {... }
}
```

Reply:
this component only works w/ the `items` prop, so it should not also own the Workspace-level fragment w/ all of the filtering & limit arguments - it doesn't concern itself w/ any of that, it just takes the items passed in, whatever they are, and renders them